R/hamcrest.R

```diff
@@ -1,162 +1,341 @@
+#' evaluates does not evaluate to \code{FALSE}, then an error condition is
+#' raised.
+#'
+#' @param value a vector with any length.
+#' @details
+#' The \code{\link{identical}} function is used to check this condition,
+#' so this function will also raise an error condition if \code{value} evaluates
+#' to \code{0} or if the result of the evaluation has any attributes.
+#' @examples \dontrun{
+#' assertFalse(is.character(1L))
+#' }
+#' @seealso
+#' \itemize{
+#' \item \code{\link{assertThat}}
+#' \item \code{\link{assertTrue}}
+#' \item \code{\link{isFalse}}
+#' }
+#' @export
 assertFalse <- function(value) {
 
 	call <- match.call()
 
 	if(!identical(value, FALSE)) {
-		stop(sprintf("\nassertFalse(%s) failed\nGot: %s", 
+		stop(sprintf("\nassertFalse(%s) failed\nGot: %s",
 				deparse0(call$value), deparse0(value)))
-	}	
-}
-
-
-
-# --------------------------------------
-# MATCHER FUNCTIONS
-# --------------------------------------
-compareReal <- function(actual, expected, tol) {
-  rel.diff <- abs(expected - actual) / abs(expected)
-  finite <- is.finite(rel.diff) & expected != 0
-  finiteValuesCloseEnough <- all(rel.diff[finite] < tol)
-  nonFiniteValuesIdentical <- identical(expected[!finite], actual[!finite])
-  
-  return( finiteValuesCloseEnough && nonFiniteValuesIdentical )
-}
-
-identical.attributes <- function(actual, expected, tol = NULL) {
-    # Should have the same set of names,
-    # though not necessarily in the same order
-    if(length(setdiff(names(expected), names(actual))) > 0) {
-        return(FALSE)
-    }
-    
-    # Otherwise verify that the values are identical
-    for(a in names(expected)) {
-        if(!identical.rec(actual[[a]], expected[[a]], tol)) {
-            return(FALSE)
-        }
-    }
-    return(TRUE)
-}
-
-identical.rec <- function(actual, expected, tol = NULL) {
-    if (length(actual) != length(expected))
-      return(FALSE)
-    if (typeof(actual) != typeof(expected))
-      return(FALSE)
-    if (!identical.attributes(attributes(actual), attributes(expected), tol)) {
-      return(FALSE)
-    }
-    if (is.list(actual)) {
-      for (i in seq_along(actual)) {
-        isSame <- identical.rec(actual[[i]], expected[[i]], tol)
-        if (!isSame){
-          return(FALSE)
-        }
-      }
-      return(TRUE)
-    } else if (!is.null(tol) && is.double(actual)) {
-      compareReal(unclass(actual), unclass(expected), tol)
-    } else if (!is.null(tol) && is.complex(actual)) {
-      compareReal(unclass(Re(actual)), unclass(Re(expected)), tol) &&
-        compareReal(unclass(Im(actual)), unclass(Im(expected)), tol)
-    } else {
-      return(identical(actual, expected))
-    }
+	}
 }
 
+### ----------------------------------------------------------------- ###
+### MATCHERS ----
+### ----------------------------------------------------------------- ###
 
+#' Checks if the expected value is close to a value within some tolerance level
```

Comment:
Should be "returns a function that checks if its *argument* is close to an *expected* value within the given tolerance."